File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
	int n, m, i, j;
	scanf("%d %d", &n, &m);
	int* arr = (int*)malloc(sizeof(int) * (n + 1));
	for (int i = 0; i < n + 1; i++) {
		arr[i] = i;
	}
	while (m--) {
		scanf("%d %d", &i, &j);
		while (i < j) {
			int tmp = arr[i];
			arr[i] = arr[j];
			arr[j] = tmp;
			i++;
			j--;
		}
	}
	for (int i = 1; i < n + 1; i++) {
		printf("%d ", arr[i]);
	}
	free(arr);
	return 0; 
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
n, m = map(int, input().split())
arr = [0] * (n + 1)
for i in range(n + 1):
    arr[i] = i
while(m > 0):
    i, j = map(int, input().split())
    while(i < j):
        tmp = arr[i]
        arr[i] = arr[j]
        arr[j] = tmp
        i += 1
        j -= 1
    m -= 1
for i in range(1, n+1):
    print(arr[i], end=" ")